proposals/0540-vision-for-layout-conformance-parity.md

```diff
@@ -0,0 +1,238 @@
+
+## Validating conformance
+
+We know about some subset of conformance issues, but there are likely to be more, and as we add new capabilities, we must be systematic in ensuring a correct implementation. Browsermakers share a set of tests for validating conformance called [WPT (Web Platform Tests)](https://github.com/web-platform-tests/wpt/tree/master/css). These have already been leveraged by React Native for pointer events, by porting the HTML to React Native. An alternative to porting the code may be to rely on Yoga’s existing capabilities of generating unit test derived from how Chrome lays out HTML fixtures.
+
+## Specific W3C conformance issues
+
+While we are likely to discover more conformance issues (and there are pending issues and PRs on GitHub that should be examined), the below list targets some of the issues that have been observed by framework authors, and individual developers:
+
+1. **`UseLegacyStretchBehaviour`:** A Yoga configuration flag set by react-native for bug compatibility, which may incorrectly stretch an items flex-basis, even when told not to grow. It has been a source of multiple reports of different behavior between browsers and RN, and does not require additional implementation within the layout engine.
+2. **Stale flex basis:** Yoga does not correctly invalidate flex basis after it changes. A fix for this was added, gated behind `YGExperimentalWebFlexBasis`, but there are concerns over its correctness. A long running issue with text truncation in a large Meta application was root-caused to this issue. An existing [pull request](https://github.com/facebook/yoga/pull/1017) may offer a fix.
+3. **Box-sizing:** While not strictly related to Yoga, react-native offers a box-sizing similar, but reportedly not quite in line with `box-sizing: 'border-box'`. More investigation is needed into the exact behavior.
+4. **Two-pass flexible item resolution:** Yoga [deviates](https://github.com/facebook/yoga/blob/5dd33acc912711c599250e3533b07c732de503ee/yoga/Yoga.cpp#L2365) from the W3C spec on how the lengths of flexible items are resolved, implementing a simpler/more performant algorithm.
+5. **Content-based minimum size:** Yoga intentionally [does not](https://github.com/facebook/yoga/blob/5dd33acc912711c599250e3533b07c732de503ee/yoga/Yoga.cpp#L2662-L2665) assign a minimum size to flexible items in deviation from the flexbox spec, for performance reasons.
+6. **Min/Max in flexible items:** At least according to [in-code documentation](https://github.com/facebook/yoga/blob/5dd33acc912711c599250e3533b07c732de503ee/yoga/Yoga.cpp#L2666-L2667), Yoga does not respect min/max size constraints on the main axis for flexible items. Because of a lack of support this is unlikely to affect existing RN code, but affects code ported from web.
```

Comment:
Interesting - I was wondering how it only used two layout passes. In the spec with min/max sizes, it's between 1 to N iterations to properly size the items